### DISCUSSION TITLE
Webpack reported a parsing error in alpine@3.0.1

### DISCUSSION THREAD

**Discussion by donaldww** [alpinejs/alpine#1439] [7. Bugs (archived)]
The following error message came from the startup routine of a phoenix application. The problem seems to be the '?' right after x_undolf.

$ mix phx.server

ERROR in ./node_modules/alpinejs/dist/module.esm.js 3010:32
Module parse failed: Unexpected token (3010:32)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders

``|     return clone2;``
``|   };``
``>   let hide = () => el._x_undoIf?.() || delete el._x_undoIf;``
``|   effect3(() => evaluate2((value) => {``
``|     value ? show() : hide();``

**Comment by KevinBatdorf** [comment]
Check whether you need to update preset-env or webpack or something in your build workflow so that it will parse the new JS optional method chaining that is showing as unexpected there.

**Reply by lukashambsch** [reply]
I'm getting the exact same error with "webpack": "^4.46.0", "@babel/core": "^7.0.0", "@babel/preset-env": "^7.0.0", and "babel-loader": "^8.0.0". Strangely, when I remove the alpine import and use the optional chaining operator in my code, I don't receive the error.

**Comment by Pulstarter** [comment]
Same problem here with webpack 5.38.1

**Comment by mrmurphy** [comment]
I just made sure babel and babel preset-env were both up-to-date. I don't think the optional chaining is what's causing the issue here. Because in my own code bundle I pasted an optional chain, and the build works fine:

```
import "../css/app.css";

const dogName = {}.dog?.name;

// import Alpine from "alpinejs";
// window.Alpine = Alpine;
// Alpine.start();
```

When I comment the Alpine import back in again, I get the same error message as above:

```
RROR in ./node_modules/alpinejs/dist/module.esm.js 3010:32
Module parse failed: Unexpected token (3010:32)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|     return clone2;
|   };
>   let hide = () => el._x_undoIf?.() || delete el._x_undoIf;
|   effect3(() => evaluate2((value) => {
|     value ? show() : hide();
 @ ./js/app.js 8:0-30 9:16-22 10:0-6
 @ multi ./js/app.js
```

**Reply by Pulstarter** [reply]
Well, the problem seems to be that V3 does not compile with webpack 4. I have upgraded to webpack 5 and all works fine now.

**Comment by danddanddand** [comment]
For anyone having trouble with webpack bundle, make sure you have defer on script tag. 

```
<script type="text/javascript" src="/your-webpack-bundle.js" defer></script>
```

This took us way longer then it should to figure out since Alpine will just fail dependent on speed DOM loads. We also use snowpack for development (which uses module and is automatically deferred) so it felt like Webpack problem bundling Alpine but wasn't.
